src/lumigo_tracer/utils.py

```diff
@@ -161,3 +162,62 @@ def is_aws_environment():
     :return: heuristically determine rather we're running on an aws environment.
     """
     return bool(os.environ.get("LAMBDA_RUNTIME_DIR"))
+
+
+class Frame:
+    MAX_VAR_LEN = 200
+
+    total_frames_size: int = 0
+
+    def __init__(self, frame_info: inspect.FrameInfo):
+        self.lineno: int = frame_info.lineno
+        self.file_name: str = frame_info.filename
+        self.function: str = frame_info.function
+        Frame.total_frames_size += len(str(self.lineno)) + len(self.file_name) + len(self.function)
+        self.variables: FrameVariables = Frame._truncate_locals(frame_info.frame.f_locals)
+
+    def to_dict(self) -> Dict[str, Union[int, str, FrameVariables]]:
+        return {
+            "lineno": self.lineno,
+            "file_name": self.file_name,
+            "function": self.function,
+            "variables": self.variables,
+        }
+
+    @staticmethod
+    def _truncate_var(var: Any) -> str:
+        var = str(var)
+        if len(var) > Frame.MAX_VAR_LEN:
+            return var[: Frame.MAX_VAR_LEN] + "..."
+        return var
+
+    @staticmethod
+    def _truncate_locals(f_locals: Dict[str, Any]) -> FrameVariables:
+        """
+        Truncate variable part or the entire variable in order to avoid exceeding the maximum frames size.
+        :param f_locals: inspect.FrameInfo.frame.f_locals
+        """
+        locals_truncated: FrameVariables = {}
+        for var_name, var_value in f_locals.items():
+            if not callable(var_value):
```

Comment:
why?
you ignore here classes, which may be really not fun when they use the factory design pattern or something like that...

Reply:
I wanted to avoid sending function, but it doesn't really matter.